home/migrations/0002_test_data.py

```diff
@@ -0,0 +1,36 @@
+from django.db import migrations, transaction
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('home', '0001_initial'),
+    ]
+
+    def generate_data(apps, schema_editor):
+        from home.models import Profile
+        user_test_data = [
+            ('Rebecca', 'RebeccaRebecca', 'Rebecca@gmail.com', 'F'),
+            ('Danit', 'DanitDanit', 'Danit@gmail.com', 'F'),
+            ('Aviv', 'AvivAviv', 'Aviv@gmail.com', 'M'),
+            ('Ido', 'IdoIdo', 'Ido@gmail.com', None),
+        ]
+
+        with transaction.atomic():
+            # Test: Create a User and Profile from a User via username.
+            for username, password, email, gender in user_test_data:
+                user = User(username=username, password=password, email=email).save()
+                if gender:
+                    Profile(user=user, gender=gender).save()
+                else:
+                    Profile(user=user).save()
```

Comment:
If you want to create the profile whether there is gender or not, you can remove the if statement.
`Profile(user=user, gender=gender).save()`
when gender is None, it the same as 
`Profile(user=user).save()`

Reply:
As discussed elsewhere, it seems that having `gender=None`is not quite the same as not setting it (The first tries to explicitly set a NULL value on the field while the 2nd uses the defined default).